jmxtrans-output/jmxtrans-output-logback/src/main/java/com/googlecode/jmxtrans/model/output/KeyOutWriter.java

```diff
@@ -147,21 +147,52 @@ public void validateSetup(Server server, Query query) throws ValidationException
 	Logger getLogger(String outputFile) {
 		return loggers.get(outputFile);
 	}
+	
+	/**
+	 * Simple class which extends the Writer Class in order to use
+	 * our logger with the GraphiteWriter2 implementation.
+	 * @author a690062
+	 *
+	 */
+	public class LogWriter extends Writer{
+		
+		private Logger logger;
+		private String delimiter;
+		
+		public LogWriter(Logger logger, String delimiter) {
+			this.logger = logger;
+			this.delimiter = delimiter;
+		}
+		
+		@Override
+		public void write(String str) throws IOException{
+			logger.info(str.replace("\n", "").replace(" ", delimiter));
+		}
+		
+		@Override
+		public void write(char[] cbuf, int off, int len) throws IOException {
+			logger.info(String.copyValueOf(cbuf, off, len).replace("\n", "").replace(" ", delimiter));
+		}
+		
+		@Override
+		public void flush() throws IOException {
+			//Do nothing			
+		}
+		
+		@Override
+		public void close() throws IOException {
+			// Do nothing
+		}
+	}
 
 	/**
-	 * The meat of the output. Very similar to GraphiteWriter.
+	 * The meat of the output. Reuses the GraphiteWriter2 class but writes in a logfile
+	 * instead of a network socket.
 	 */
 	@Override
 	public void internalWrite(Server server, Query query, ImmutableList<Result> results) throws Exception {
-		List<String> typeNames = getTypeNames();
-
-		for (Result result : results) {
-			if (isNumeric(result.getValue())) {
-
-				logger.info(KeyUtils.getKeyString(server, query, result, typeNames, null) + delimiter
-						+ result.getValue().toString() + delimiter + result.getEpoch());
-			}
-		}
+		LogWriter logwriter = new LogWriter(logger,delimiter);
```

Comment:
This logwriter could be a field, no need to instantiate a new one for each message.